server/src/main/java/org/opensearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -536,21 +560,24 @@ public void handleException(TransportException exp) {
                         onCompletionListener.onResponse(response);
                     }, e -> handleException(primaryShardReference, e));
 
+                    final Replicas<ReplicaRequest> replicasProxy = newReplicasProxy();
+                    final IndexShard indexShard = primaryShardReference.indexShard;
+                    final Replicas<ReplicaRequest> termValidationProxy = primaryTermValidationReplicasProxy();
+
                     new ReplicationOperation<>(
                         primaryRequest.getRequest(),
                         primaryShardReference,
                         ActionListener.map(responseListener, result -> result.finalResponseIfSuccessful),
-                        newReplicasProxy(),
+                        replicasProxy,
                         logger,
                         threadPool,
                         actionName,
                         primaryRequest.getPrimaryTerm(),
                         initialRetryBackoffBound,
                         retryTimeout,
-                        ReplicationProxyFactory.create(
-                            primaryShardReference.indexShard,
-                            getReplicationMode(primaryShardReference.indexShard)
-                        )
+                        indexShard.isRemoteTranslogEnabled()
+                            ? new ReplicationModeAwareProxy<>(getReplicationMode(indexShard), replicasProxy, termValidationProxy)
+                            : new FanoutReplicationProxy<>(replicasProxy)
```

Comment:
Can we get rid of `ReplicationProxyFactory` altogether?

Reply:
Done already